### PR TITLE
Module 8

### PR DESCRIPTION
**Pull request by JoeBawn**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,11 +6,27 @@ env:
...
+  - secure: "d4NtPyHgynqStDDrE/vBG3BWAM4fJn4x5f7fnfwLmanooYrtC2zd7EAIfTBTnF0e3rgJDJt+3MeLjIlOlBxJddDboUtcQkkEwFayhwnHEvxCnRyPS9oLMmRsmqiSULCA9U5sFtilVRHGDejCWKpoWUagVvzLjga1MlL/dweUR24046VjD0m6javl0i0a0GED9RxARGJo0vlj31jOXJu9RDi2w+0ppkCOxcYeH9+Dzt02yhrpX0s9DTskZzDw4JdUym7W6fA0e06yK9MEOshMgfQOokE1qt3X6fJFKSvds1dnXOrL6+2JyvNyiuc/JRDujP6+Y+whJhhg2hKF91AU/udenT8VdOPQEY94yK3zVtebjjwv08iIZdJ7pGOp76qb3EGZV7rN2kjzum7Ay7sFhmtoTcTPrO/1oIjf4eCxIOPWMr21njPmsUrufBxAjEMuB8M/HR6y9ZPhm+wpHbUU9wD0YhNLS1j9i0sCEaUF6fJMDcPDIHBmvsfGGIN7VTRaMwvgfeFftDakH2Z8Ore8ZZ6j/lxytcbXu4n/pl/JJUXbF8+bw6GZrAguLPrh0qbqQTPJXSzHKBh53+1nTh5s63PDXZ4XXgT9v/ghMyW4AyxNosRvmqjrPAihQiAjSPqKebP1gBJgPJH6WyNLJpnKBirewgulus1t8Q0EVfWSUS8="
 
 script:
 - docker build --tag todo-app:test --target test .
 - docker run todo-app:test todo_app/Tests/test_items.py
 - docker run --env-file .env.test todo-app:test todo_app/Tests/test_app.py
 - docker run --env TRELLO_API_BOARD_ID --env TRELLO_API_TOKEN --env TRELLO_API_KEY  todo-app:test todo_app/Selenium_Tests/test_Selenium.py
+- echo $DOCKER_PASSWORD | docker login --username $DOCKER_USER --password-stdin
```
Comment (hugh-emerson) [inline .travis.yml:18]: Slightly odd to have this up here - why not move it into before_deploy next to where it's relevant?

File: .travis.yml [hunk 2]
```diff
@@ -6,11 +6,27 @@ env:
...
 
+before_deploy:
+- heroku container:login
+- docker build --target production --tag $DOCKER_USER/todo-app:latest --target production .
+- docker push $DOCKER_USER/todo-app:latest
+- docker tag $DOCKER_USER/todo-app:latest registry.heroku.com/todoapp-jb/web
```
Comment (hugh-emerson) [inline .travis.yml:24]: You could also just create both tags in the first place with e.g. `docker build --tag tag-one --tag tag-two .`

File: .travis.yml [hunk 3]
```diff
@@ -6,11 +6,27 @@ env:
...
 script:
 - docker build --tag todo-app:test --target test .
 - docker run todo-app:test todo_app/Tests/test_items.py
 - docker run --env-file .env.test todo-app:test todo_app/Tests/test_app.py
 - docker run --env TRELLO_API_BOARD_ID --env TRELLO_API_TOKEN --env TRELLO_API_KEY  todo-app:test todo_app/Selenium_Tests/test_Selenium.py
+- echo $DOCKER_PASSWORD | docker login --username $DOCKER_USER --password-stdin
 
+before_deploy:
+- heroku container:login
+- docker build --target production --tag $DOCKER_USER/todo-app:latest --target production .
```
Comment (hugh-emerson) [inline .travis.yml:22]: You've accidentally written `--target production` twice

**Review by hugh-emerson — APPROVED** [review]
Looks good and I see your app running successfully on Heroku 👍 

Merge when you're ready

File: .travis.yml (hunk 1 above)
Comment (JoeBawn) [reply .travis.yml]: Moved to before_deploy

File: .travis.yml (hunk 3 above)
Comment (JoeBawn) [reply .travis.yml]: removed duplicate --target production

**Pull request merged**